flaky test: //kythe/go/util/riegeli:riegeli_test

Looks like 1 test fails on `bazel //...` on macOS

```
Executing tests from //kythe/go/util/riegeli:riegeli_test
-----------------------------------------------------------------------------
2018/11/12 20:55:10 Compressed size of "TestReadWriteProto/zstd": 218225 bytes
2018/11/12 20:55:10 Compressed size of "TestReadWriteProto/uncompressed": 723013 bytes
2018/11/12 20:55:10 Compressed size of "TestReadWriteProto/brotli": 55538 bytes
2018/11/12 20:55:10 Compressed size of "TestReadWriteProto/default": 55537 bytes
2018/11/12 20:55:11 Compressed size of "TestReadWriteProto/brotli,transpose": 32529 bytes
2018/11/12 20:55:11 Compressed size of "TestReadWriteProto/uncompressed,transpose": 581299 bytes
2018/11/12 20:55:13 Compressed size of "TestReadWriteProto/transpose": 32522 bytes
--- FAIL: TestReadWriteProto (0.00s)
    --- FAIL: TestReadWriteProto/zstd (0.65s)
        riegeli_test.go:204: Read error: proto: can't skip unknown wire type 7
FAIL
```

--- a/kythe/go/util/riegeli/compression.go
+++ b/kythe/go/util/riegeli/compression.go
@@ -76,11 +76,33 @@ func newCompressor(opts *WriterOptions) (compressor, error) {
 		w := cbrotli.NewWriter(buf, brotliOpts)
 		return &sizePrefixedWriterTo{buf: buf, WriteCloser: w}, nil
 	case zstdCompression:
-		w := zstd.NewWriterLevel(buf, opts.compressionLevel())
-		return &sizePrefixedWriterTo{buf: buf, WriteCloser: w}, nil
+		lvl := opts.compressionLevel()
+		return &sizePrefixedWriterTo{buf: buf, WriteCloser: &batchCompressor{
+			Compress: func(src []byte) ([]byte, error) { return zstd.CompressLevel(nil, src, lvl) },
+			Buffer:   buf,
+		}}, nil
 	default:
 		return nil, fmt.Errorf("unsupported compression_type: '%s'", []byte{byte(opts.compressionType())})
 	}
+}
+
+// A batchCompressor buffers all bytes written to it.  On Close, the Buffer is compressed.
+type batchCompressor struct {
+	Compress func([]byte) ([]byte, error)
+	Buffer   *bytes.Buffer
+}
+
+// Write implements part of the io.WriteCloser interface.
+func (b *batchCompressor) Write(buf []byte) (int, error) { return b.Buffer.Write(buf) }
+
+// Close implements part of the io.WriteCloser interface.
+func (b *batchCompressor) Close() error {
+	compressed, err := b.Compress(b.Buffer.Bytes())
+	if err != nil {
+		return err
+	}
+	*b.Buffer = *bytes.NewBuffer(compressed)
+	return nil
 }
 
 type sizePrefixedWriterTo struct {

